app/model/lib/chart.py

```diff
@@ -98,15 +106,35 @@ def to_html(self):
             right_units_label = f"ln({right_units_label})"
 
         for (df, label) in converted_data_left:
-            scatter_params = self._get_scatter_params(df, label)
+            scatter_params = self._get_scatter_params(df, label, log=self.log_left)
             fig.add_trace(go.Scatter(**scatter_params), secondary_y=False)
 
         for (df, label) in converted_data_right:
-            scatter_params = self._get_scatter_params(df, label)
+            scatter_params = self._get_scatter_params(df, label, log=self.log_right)
             scatter_params = dict(**scatter_params, line={'dash': 'dot'})
 
             fig.add_trace(go.Scatter(**scatter_params), secondary_y=True)
 
+        xaxis_range       = self._calculate_x_range(converted_data_left + converted_data_right)
+        left_yaxis_range  = self._calculate_y_range(converted_data_left)
+        right_yaxis_range = self._calculate_y_range(converted_data_right)
+
+        for index, (x0, x1, label, text) in enumerate(self.regions):
+            y0, y1 = left_yaxis_range
+
+            fig.add_trace(
+                go.Scatter(
+                    name=label,
+                    x=[x0, x0, x1, x1, x0],
+                    y=[y0, y0, y0, y1, y1],
+                    opacity=0.15,
+                    line_width=0,
+                    fill="toself",
+                    hovertemplate=text,
+                    mode="text",
+                ),
+            )
```

Comment:
There are a number of ways to highlight a particular region on a plotly chart, but drawing a "trace" that forms a self-closing shape is the only one I found that lets us actually put it on the legend.